### PR TITLE
feat(vedbokarvottord): New application

### PR DESCRIPTION
**Pull request by johannaagma**
## What

New application type for mortgage certificate

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by jonbjarnio — APPROVED** [review]

**Review by thordurhhh — APPROVED** [review]
Assets service LGTM 👍

File: libs/api/domains/assets/src/lib/api-domains-assets.service.ts
Comment (saevarma) [inline libs/api/domains/assets/src/lib/api-domains-assets.service.ts:254]: This will result in Promise array for `properties` you need to use Promise.all to resolve
```suggestion
        properties: await Promise.all(getRealEstatesRes.properties.map(
```

If the properties list can be large you might need to look into batching this.

**Review by saevarma — CHANGES_REQUESTED** [review]
Minor comments in Core owned files.
Also noticed promise issue in assets service.

File: libs/clients/syslumenn/src/clientConfig.json
Comment (johannaagma) [reply libs/clients/syslumenn/src/clientConfig.json:1073]: This is not from me, this is autogenerated from the syslumenn API swagger :)

File: libs/api/domains/mortgage-certificate/src/lib/graphql/main.resolver.ts
Comment (johannaagma) [reply libs/api/domains/mortgage-certificate/src/lib/graphql/main.resolver.ts:15]: I added 5 more spaces in front of "@island-is/..", since "/libs/application/template-api-modules/src/lib/modules/templates/mortgage-certificate-submission" is a long line, and I wanted "@island-is/.." to remain in a straight line. Do you want me to change it back?

Then it might look something like this:

image


Instead of 

image

File: workspace.json
Comment (johannaagma) [reply workspace.json:3718]: Because Styrmir from Sýslumenn had not updated the Test API so it defined the response schema correctly for the endpoint /VedbokavottordRegluverk.

But I will change this back and ask Styrmir to update their Test service asap

File: .github/CODEOWNERS
Comment (saevarma) [reply .github/CODEOWNERS:9]: Ahh I see, no this is good :)

File: workspace.json
Comment (saevarma) [reply workspace.json]: I see, guessed that it might be the reason. But I think we should try to not depend on third party dev env as it might not be as stable as their test env.

**Review by saevarma — APPROVED** [review]
Core files looks good!

File: libs/clients/syslumenn/src/lib/syslumennClient.service.ts
Comment (RunarVestmann) [inline libs/clients/syslumenn/src/lib/syslumennClient.service.ts:285]: Perhaps we should remove this commented code

**Review by RunarVestmann — APPROVED** [review]
One minor comment regarding commented code and also I'd like to mention that it would've been nice to see some tests added to the client for the new methods.

However I will not block this PR, I approve of the changes made to the files that Stefna is a code owner of.

**Comment by johannaagma** [comment]
@RunarVestmann There are few more tweaks we need to do for this application, I will add to that TODO list to add tests for the newly added functions in SyslumennClient :)

**Review by obmagnusson — APPROVED** [review]

**Review by albinagu — APPROVED** [review]

**Pull request merged**
